concepts/collect/about.md

```diff
@@ -1,3 +1,52 @@
 # About
 
-TODO
+To understand the `collect` function it's important to know that it is used to resolve an iterator. Recall that an iterator is a sequence of items
+that can be chained with other iterators. Iterators are _lazy_, which means they do not execute until they are resolved by a
+[consuming adaptor][consuming adaptor]. Consuming adaptors process all items in the iterator to produce some kind of output.
```

Comment:
> they do not execute until they are resolved by a consuming adaptor

I think this is misleading. Consuming adapters are one way to resolve an iterator, but there are others, such as a `for` loop:

```rust
for ch in "hello world".chars() {
  print!("{}", ch);
}
```

Reply:
Yes, that's a good point. I'll finesse it.